src/devnet/create-devnet-wrapper.ts

```diff
@@ -40,10 +40,17 @@ export function createIntegratedDevnet(hre: HardhatRuntimeEnvironment): Integrat
         );
     }
 
+    // Check CPU architecture
+    const arch = process.arch;
+    let tag = devnetNetwork.dockerizedVersion || DEFAULT_DEVNET_DOCKER_IMAGE_TAG;
+    if (arch.includes("arm")) {
```

Comment:
1) Our images are arm64 compatible. I don't know if that also means they are arm compatible - investigate this. Also, instead of using `arch.includes`, rather use something like `["arm", "arm64"].includes` (although, depending on your findings, this might change to just checking if `arch === "arm64"`).
2) Extract the arch checking and appending of the suffix to a util function.
3) Don't use `tag.includes` but `tag.endsWith`.

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/123/commits/3cef8f790f91f05e5332c46e3f7cced0ec208fe0